Library/MeterGeometry.cpp

```diff
@@ -0,0 +1,370 @@
+/* Copyright (C) 2016 Rainmeter Project Developers
+ *
+ * This Source Code Form is subject to the terms of the GNU General Public
+ * License; either version 2 of the License, or (at your option) any later
+ * version. If a copy of the GPL was not distributed with this file, You can
+ * obtain one at <https://www.gnu.org/licenses/gpl-2.0.html>. */
+
+#include "StdAfx.h"
+#include "MeterGeometry.h"
+#include "Logger.h"
+
+D2D1_COLOR_F ToColorF(const Gdiplus::Color& color)
+{
+	return D2D1::ColorF(color.GetR() / 255.0f, color.GetG() / 255.0f, color.GetB() / 255.0f, color.GetA() / 255.0f);
+}
+
+MeterGeometry::MeterGeometry(Skin* skin, const WCHAR* name) : Meter(skin, name),
+	m_Shapes(),
+	m_MeasureOptions(),
+	m_NeedsRedraw(false)
+{
+}
+
+MeterGeometry::~MeterGeometry()
+{
+}
+
+void MeterGeometry::Initialize()
+{
+	Meter::Initialize();
+}
+
+bool MeterGeometry::ParseShape(GeometryShape & shape, const LPCWSTR & optionName, const LPCWSTR & optionValue)
+{
+	if (_wcsicmp(optionName, L"Rectangle") == 0) shape.m_Shape = ParseRect(shape, ConfigParser::ParseRECT(optionValue));
+	else return false;
+	//Easy to do with a rectangle, but this will calculate the bounds of all shape types
+	if (shape.m_Shape) {
```

Comment:
Could we separate this bit out into another function like `UpdateSize()` or something?